lib/chf/synthetic_category.rb

```diff
@@ -82,15 +91,58 @@ def self.keys
       definitions.keys
     end
 
+    def self.all
+      keys.collect { |key| self.new(key) }
+    end
+
+    def self.from_slug(slug)
+      if slug.blank?
+        nil
+      elsif has_key?(slug)
+        self.new(slug)
+      elsif has_key?(slug.underscore)
+        self.new(slug.underscore)
+      else
+        nil
+      end
+    end
+
+
     attr_accessor :category_key
 
+    def slug
```

Comment:
Hmm....  I decided it was better for SEO and general look of professionalism to have the url look like `/focus/portraits-and-people` instead of `/focus/portraits_and_people`.  But it's difficult to work with symbols with dashes in them in ruby source code. So I don't know if it's truly _neccesary_, it just seemed convenient to me. What do you think?